Actor.cs

```diff
@@ -0,0 +1,85 @@
+internal class Actor : MonoBehaviour
+{
+	private static List<Actor> _actors = new List<Actor>();
+	public int StartingHealth;
+	int _currentHealth;
+	private bool _isDead;
+	private CancellationTokenSource _healthCancellationTokenSource;
+	private CancellationTokenSource _deathCancellationTokenSource;
+
+	public event Action<Actor> ActorDied;
```

Comment:
You probably also want `ActorDying` & `ActorHealthChanging` events
These would occur because the change of the backing data

You also likely want the event payloads to include the related data, in the case of `ActorHealthChanging` that would be something like `previousHealth, newHealth`